sympy/utilities/lambdify.py

```diff
@@ -450,6 +455,45 @@ def wrapper(*argsx, **kwargsx):
                     "Expression:\n\n{expr}").format(sig=sig, expr=expr_str)
     return func
 
+def should_dummify(symbol, printer=None):
+    """
+    Returns whether ``symbol`` should be dummified.
+
+    Parameter
+    ---------
+    symbol : Represents the symbol for which dummify should be checked
+    printer : Represents the printer provided by lambdify function or by the user
+
+    Examples
+    ========
+
+    >>> from sympy.utilities.lambdify import should_dummify
+    >>> from sympy.abc import x
+    >>> should_dummify('elif')
+    True
+    >>> should_dummify(x)
+    False
+
+    """
+    if printer is not None:
+        if inspect.isfunction(printer):
+            lambdarepr = printer
+        else:
+            if inspect.isclass(printer):
+                lambdarepr = lambda expr: printer().doprint(expr)
+            else:
+                lambdarepr = lambda expr: printer.doprint(expr)
+    else:
+        #XXX: This has to be done here because of circular imports
+        from sympy.printing.lambdarepr import lambdarepr
+
+    if keyword.iskeyword(lambdarepr(symbol)):
+        return True
+    else:
+        valid_identifier_regex = re.compile('^[^\d\W]\w*\Z', re.UNICODE)
```

Comment:
By looking at the tests of ``lambdarepr``, it does not seem that it returns unicode strings, at least not in the tests. Independently, the usage of re.compile does not make here, since re.compile is called for every ``should_dummify`` function call. Therefore, you can either use re.match directly here or cache the compiled regex (line 493).